salt/modules/dnsmasq.py

```diff
@@ -140,20 +151,37 @@ def get_config(config_file="/etc/dnsmasq.conf"):
         salt '*' dnsmasq.get_config config_file=/etc/dnsmasq.conf
     """
     dnsopts = _parse_dnamasq(config_file)
-    if "conf-dir" in dnsopts:
-        for filename in os.listdir(dnsopts["conf-dir"]):
-            if filename.startswith("."):
-                continue
-            if filename.endswith("~"):
-                continue
-            if filename.endswith("#") and filename.endswith("#"):
-                continue
-            dnsopts.update(
-                _parse_dnamasq("{}/{}".format(dnsopts["conf-dir"], filename))
-            )
+    if follow:
+        for config in _discover_configs(dnsopts):
+            dnsopts.update(_parse_dnamasq(config))
     return dnsopts
 
 
+def _discover_configs(dnsopts):
+    if "conf-dir" not in dnsopts:
+        return []
+
+    config_files = []
+    # See "--conf-dir" in manpage
+    config_dir_and_maybe_glob = dnsopts["conf-dir"].split(",")
+    config_dir = config_dir_and_maybe_glob[0]
+    try:
+        config_file_glob = config_dir_and_maybe_glob[1]
+    except IndexError:
+        config_file_glob = "*"
+
+    for filename in glob(os.path.join(config_dir, config_file_glob)):
+        if filename.startswith("."):
+            continue
+        if filename.endswith("~"):
+            continue
+        if filename.endswith("#") and filename.endswith("#"):
```

Comment:
This is checking the same value 2 times which is probably not what is intended.

Reply:
done in https://github.com/saltstack/salt/pull/65215/commits/227c3c6c38bd4dce38a707bfca718fe7b8864949